src/kubectlUtils.ts

```diff
@@ -1,344 +1,369 @@
+    }
+
+    return deploymentName;
+}
+
+/**
+ * Query the pod list for the specified label.
+ *
+ * @param kubectl the kubectl client.
+ * @param labelQuery the query label.
+ * @return the pod list.
+ */
+export async function findPodsByLabel(kubectl: Kubectl, labelQuery: string): Promise<KubernetesCollection<Pod>> {
+    const getResult = await kubectl.asJson<KubernetesCollection<Pod>>(`get pods -o json -l ${labelQuery}`);
+    if (failed(getResult)) {
+        throw new Error('Kubectl command failed: ' + getResult.error[0]);
+    }
+    return getResult.result;
+}
+
+/**
+ * Wait and block until the specified pod's status becomes running.
+ *
+ * @param kubectl the kubectl client.
+ * @param podName the pod name.
+ */
+export async function waitForRunningPod(kubectl: Kubectl, podName: string): Promise<void> {
+    while (true) {
+        const shellResult = await kubectl.invokeAsync(`get pod/${podName} --no-headers`);
+        if (shellResult.code !== 0) {
+            throw new Error(`Failed to get pod status: ${shellResult.stderr}`);
+        }
+        const status = shellResult.stdout.split(/\s+/)[2];
+        kubeChannel.showOutput(`pod/${podName} status: ${status}`);
+        if (status === "Running") {
+            return;
+        } else if (!isTransientPodState(status)) {
+            const logsResult = await kubectl.invokeAsync(`logs pod/${podName}`);
+            kubeChannel.showOutput(`Failed to start the pod "${podName}". Its status is "${status}".
+                Pod logs:\n${logsResult.code === 0 ? logsResult.stdout : logsResult.stderr}`);
+            throw new Error(`Failed to start the pod "${podName}". Its status is "${status}".`);
+        }
+
+        await sleep(1000);
+    }
+}
+
+function isTransientPodState(status: string): boolean {
+    return status === "ContainerCreating" || status === "Pending" || status === "Succeeded";
+}
+
+/**
+ * Get the specified resource information.
+ *
+ * @param kubectl the kubectl client.
+ * @param resourceId the resource id.
+ * @return the result as a json object, or undefined if errors happen.
+ */
+export async function getResourceAsJson<T extends KubernetesResource | KubernetesCollection<KubernetesResource>>(kubectl: Kubectl, resourceId: string, resourceNamespace?: string): Promise<T | undefined> {
+    const nsarg = resourceNamespace ? `--namespace ${resourceNamespace}` : '';
+    const shellResult = await kubectl.asJson<T>(`get ${resourceId} ${nsarg} -o json`);
+    if (failed(shellResult)) {
+        vscode.window.showErrorMessage(shellResult.error[0]);
+        return undefined;
+    }
+    return shellResult.result;
+}
+
+export async function createResourceFromUri(uri: vscode.Uri, kubectl: Kubectl) {
+    await changeResourceFromUri(uri, kubectl, 'create', 'creating', 'created');
+}
+
+export async function deleteResourceFromUri(uri: vscode.Uri, kubectl: Kubectl) {
+    const result = await vscode.window.showWarningMessage('Are you sure you?', 'Delete', 'Cancel');
```

Comment:
That confirmation message is probably not what you intended *grin*.

Reply:
done.